python/ray/tune/trial_runner.py

```diff
@@ -489,7 +494,10 @@ def _start_trial(trial: Trial) -> bool:
         if next_trial is not None:
             if _start_trial(next_trial):
                 may_handle_events = False
-            else:
+            elif next_trial.status != Trial.ERROR:
```

Comment:
quick question, is there a reason why we shouldn't just do :

```
self.trial_executor.stage_and_update_status(self._trials)
next_trial = self.trial_executor.get_staged_trial()
```
?

Reply:
This would break trial scheduling behavior. If a scheduler chooses a trial in `choose_trial_to_run()` we would expect it to be the next to start. If we just get a trial that is ready to stage we would introduce an arbitrary choice in the placement group manager. 
As an example, this would break FIFO scheduling, but it's actually not limited to this. Basically we'll lose the API guarantees of `choose_trial_to_run`.

I guess we could get rid of this fallback-launching behavior and just try to start the trial returned by the scheduler.

Reply:
I'm OK with current solution!